cors/mod.ts

```diff
@@ -0,0 +1,207 @@
+import { Drash } from "../deps.ts";
+import { vary } from "./deps.ts";
+
+interface ArrayOfValueOrArray<T> extends Array<ValueOrArray<T>> {}
+type ValueOrArray<T> = T | ArrayOfValueOrArray<T>;
+type OriginOption =
+  | ((
+    request: Drash.Http.Request,
+    response?: Drash.Http.Response,
+  ) => string | Promise<string>)
+  | ValueOrArray<string | boolean | RegExp>;
+
+type CorsMiddlewareConfig = {
+  origin?: OriginOption;
+  credentials?:
+    | ((
+      request: Drash.Http.Request,
+      response?: Drash.Http.Response,
+    ) => boolean | Promise<boolean>)
+    | boolean;
+  exposeHeaders?: string | string[];
+  allowMethods?: string | string[];
+  allowHeaders?: string | string[];
+  maxAge?: number;
+  optionsSuccessStatus?: number;
+  preflight?: boolean;
+};
+
+const isOriginAllowed = (reqOrigin: string, origin: OriginOption): boolean => {
+  if (Array.isArray(origin)) {
+    for (let i = 0; i < origin.length; ++i) {
+      if (isOriginAllowed(reqOrigin, origin[i])) {
+        return true;
+      }
+    }
+
+    return false;
+  } else if (typeof origin === "string") {
+    return reqOrigin === origin;
+  } else if (origin instanceof RegExp) {
+    return origin.test(reqOrigin);
+  } else {
+    return !!origin;
+  }
+};
+
+export const CorsMiddleware = ({
+  origin = "*",
+  credentials = false,
+  exposeHeaders,
+  allowMethods = "GET,HEAD,PUT,PATCH,POST,DELETE",
+  allowHeaders,
+  maxAge,
+  optionsSuccessStatus = 204,
+  preflight = true,
+}: CorsMiddlewareConfig = {}) =>
+  async (
+    request: Drash.Http.Request,
+    response: Drash.Http.Response,
+  ): Promise<void> => {
+    let options: any = {
```

Comment:
Can this be typed?

Reply:
https://github.com/drashland/deno-drash-middleware/pull/1/commits/1627c064bf674eeeccbe726cd79bd685da78dd8b